cpp/src/parquet/metadata.cc

```diff
@@ -826,6 +831,32 @@ class FileMetaData::FileMetaDataImpl {
     file_decryptor_ = std::move(file_decryptor);
   }
 
+  void set_column_offsets(const std::vector<std::shared_ptr<OffsetIndex>>& column_offsets, int64_t expected_num_rows) {
+    if (num_row_groups() != 1) {
+      throw ParquetException(
+              "This operation can only be applied to metadata with a single row group");
+    }
+
+    format::RowGroup& row_group = metadata_->row_groups[0];
+    int idx = 0;
+    for (format::ColumnChunk& chunk : row_group.columns) {
+      // Assume a chunk has only 1 page for now
```

Comment:
I'm actually not quite sure how this should go when a ColumnChunk has multiple pages or if that is even possible.

Reply:
a column chunk will very frequently has multiple pages.